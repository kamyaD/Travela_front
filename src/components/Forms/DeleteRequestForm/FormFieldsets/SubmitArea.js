import React from 'react';
import {PropTypes} from 'prop-types';

const SubmitArea = ({ onCancel, deleteChecklistItem, deleteReason }) => {

  return (
    <fieldset>
      <div className="submit-area">
        <button type="button" className="bg-btn bg-btn--inactive" onClick={onCancel} id="cancel">
          Cancel
        </button>
        <button 
          type="button"
          className="restore-checklist-items__footer--delete"
          id="oncancel" onClick={deleteChecklistItem}
          disabled={!deleteReason}
        >
          Disable
        </button>
      </div>
    </fieldset>
  );
};

SubmitArea.propTypes = {
  onCancel: PropTypes.func.isRequired,
  deleteChecklistItem: PropTypes.func.isRequired,
  deleteReason: PropTypes.string,
};
SubmitArea.defaultProps = {
  deleteReason: null,
};

export default SubmitArea;
