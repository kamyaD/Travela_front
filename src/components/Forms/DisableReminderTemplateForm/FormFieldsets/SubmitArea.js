import React from 'react';
import {PropTypes} from 'prop-types';

const SubmitArea = ({
  onCancel, disableEmailReminder,
  disableReminderTemplate, disableReason,
  templateReason, conditionReason, mode }) => {
  return (
    <fieldset>
      <div className="submit-area">
        <button type="button" className="bg-btn bg-btn--inactive" onClick={onCancel} id="cancel">
          Cancel
        </button>
        {
          mode === 'add' ?
            (
              <button
                type="button"
                className="restore-checklist-items__footer--delete"
                id="oncancel"
                onClick={disableReminderTemplate? disableReminderTemplate : disableEmailReminder}
                disabled={!disableReason}
              >
              Disable
              </button>
            )
            : ''
        }
      </div>
    </fieldset>
  );
};

SubmitArea.propTypes = {
  onCancel: PropTypes.func.isRequired,
  disableEmailReminder: PropTypes.func.isRequired,
  disableReminderTemplate: PropTypes.func.isRequired,
  disableReason: PropTypes.string,
  conditionReason: PropTypes.string,
  templateReason: PropTypes.string,
  mode: PropTypes.string,
};
SubmitArea.defaultProps = {
  disableReason: null,
  conditionReason:'',
  templateReason: '',
  mode: ''
};

export default SubmitArea;
