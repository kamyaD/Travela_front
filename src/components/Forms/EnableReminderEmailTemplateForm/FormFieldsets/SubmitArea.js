import React from 'react';
import {PropTypes} from 'prop-types';

const SubmitArea = ({ onCancel, enableEmailReminderTemplate }) => {

  return (
    <fieldset>
      <div className="submit-area delete-checklist-item__right">
        <button type="button" className="bg-btn bg-btn--inactive" onClick={onCancel} id="cancel">
          Cancel
        </button>
        <button 
          type="button"
          className="bg-btn bg-btn--active"
          id="oncancel" onClick={enableEmailReminderTemplate}
        >
          Enable
        </button>
      </div>
    </fieldset>
  );
};

SubmitArea.propTypes = {
  onCancel: PropTypes.func.isRequired,
  enableEmailReminderTemplate: PropTypes.func.isRequired,
};

export default SubmitArea;
