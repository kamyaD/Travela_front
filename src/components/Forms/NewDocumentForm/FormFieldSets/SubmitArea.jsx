import React from 'react';
import PropTypes from 'prop-types';

const SubmitArea = ({ onCancel, hasBlankFields }) => (
  <fieldset>
    <div className="submit-area">
      <button type="button" className="bg-btn bg-btn--inactive" id="cancel" onClick={onCancel}>
        Cancel
      </button>
      <button type="submit" className="bg-btn bg-btn--active" id="submit" disabled={hasBlankFields}>
        Add
      </button>
    </div>
  </fieldset>
);

SubmitArea.propTypes = {
  onCancel: PropTypes.func.isRequired,
  hasBlankFields: PropTypes.bool.isRequired
};

export default SubmitArea;
