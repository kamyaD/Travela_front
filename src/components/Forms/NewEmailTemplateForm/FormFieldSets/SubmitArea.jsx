import React from 'react';
import {PropTypes} from 'prop-types';
import ButtonLoadingIcon from '../../ButtonLoadingIcon';

const SubmitArea = (props) => {
  const { hasBlankFields, onCancel, send, selection, loading, reversed} = props;
  return (
    <fieldset>
      <div className={selection ? `submit-area submit-area--${selection}` : 'submit-area'}>
        {
          reversed ? (
            <React.Fragment>
              {CancelButton(onCancel)}
              {SubmitButton(hasBlankFields, loading, send)}
            </React.Fragment>
          ) : (
            <React.Fragment>
              {SubmitButton(hasBlankFields, loading, send)}
              {CancelButton(onCancel)}
            </React.Fragment>
          )
        }
      </div>
    </fieldset>
  );
};

const CancelButton = (onCancel) => (
  <button
    type="button"
    className="bg-btn bg-btn--inactive"
    onClick={onCancel}
    id="cancel">
          Cancel
  </button>
);

const SubmitButton = ( hasBlankFields, loading, send) => (
  <button
    type="submit" disabled={hasBlankFields || loading}
    className="bg-btn bg-btn--active" id="submit">
    <ButtonLoadingIcon isLoading={loading} buttonText={send} />
  </button>
);

SubmitArea.propTypes = {
  onCancel: PropTypes.func.isRequired,
  hasBlankFields: PropTypes.bool,
  send: PropTypes.string.isRequired,
  selection: PropTypes.string,
  loading: PropTypes.bool,
  reversed: PropTypes.bool,
};

SubmitArea.defaultProps = {
  hasBlankFields: false,
  reversed: false,
};

SubmitArea.defaultProps = {
  selection: '',
  loading: false,
};

export default SubmitArea;
