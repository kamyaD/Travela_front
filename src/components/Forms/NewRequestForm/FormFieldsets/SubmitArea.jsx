import React from 'react';
import {PropTypes} from 'prop-types';

const SubmitArea = (props) => {
  const { hasBlankFields, onCancel, send, cancel, modalType, onEditCancel, selection } = props;

  return (
    <fieldset>
      <div className={selection ? `submit-area submit-area--${selection}` : 'submit-area'}>
        { modalType === 'edit accomodation' ? (
          <button type="button" className="bg-btn bg-btn--inactive" id="oncancel" onClick={onEditCancel}>
          Cancel
          </button>)
          : (
            <button type="button" className="bg-btn bg-btn--inactive" onClick={onCancel} id="cancel">
              {cancel ? cancel : 'Clear'}
            </button>
          )}
        <button type="submit" disabled={hasBlankFields} className="bg-btn bg-btn--active" id="submit">
          { send }
        </button>
      </div>
    </fieldset>
  );
};

SubmitArea.propTypes = {
  onCancel: PropTypes.func.isRequired,
  hasBlankFields: PropTypes.bool.isRequired,
  send: PropTypes.string.isRequired,
  selection: PropTypes.string,
  cancel: PropTypes.string.isRequired,
  modalType: PropTypes.string.isRequired,
  onEditCancel: PropTypes.func.isRequired
};

SubmitArea.defaultProps = {
  selection: ''
};

export default SubmitArea;
