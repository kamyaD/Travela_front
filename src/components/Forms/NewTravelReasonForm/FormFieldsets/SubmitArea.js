import React from 'react';
import { PropTypes } from 'prop-types';
import ButtonLoadingIcon from '../../ButtonLoadingIcon';

const SubmitArea = ({ onCancel, hasBlankFields, travelReason }) => {
  const { isCreating } = travelReason;
  return (
    <fieldset>
      <div className="submit-area delete-checklist-item__right">
        <button
          type="button"
          className="bg-btn bg-btn--inactive"
          onClick={onCancel} id="cancel">
          Cancel
        </button>
        <button
          type="submit"
          className="bg-btn bg-btn--active"
          disabled={hasBlankFields || isCreating}
        >
          <ButtonLoadingIcon isLoading={isCreating} buttonText="Add Reason" />
        </button>
      </div>
    </fieldset>
  );
};

SubmitArea.propTypes = {
  onCancel: PropTypes.func.isRequired,
  hasBlankFields: PropTypes.bool.isRequired,
  travelReason: PropTypes.object
};

SubmitArea.defaultProps = {
  travelReason: {}
};

export default SubmitArea;
