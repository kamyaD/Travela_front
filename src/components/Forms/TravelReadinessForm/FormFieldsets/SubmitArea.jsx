import React from 'react';
import {PropTypes} from 'prop-types';
import ButtonLoadingIcon from '../../ButtonLoadingIcon';

const SubmitArea = (props) => {
  const { hasBlankFields,onCancel, send, selection, loading, documentUploaded } = props;

  return (
    <fieldset>
      <div
        className={selection ? `submit-area submit-area--${selection}` : 'submit-area'}>
        <button
          type="button"
          className="bg-btn bg-btn--inactive"
          onClick={onCancel}
          id="cancel"
          disabled={loading || documentUploaded}
        >
          Cancel
        </button>
        <button
          type="submit" disabled={hasBlankFields || loading || documentUploaded}
          className="bg-btn bg-btn--active" id="submit">
          <ButtonLoadingIcon isLoading={loading || documentUploaded} buttonText={send} />
        </button>
      </div>
    </fieldset>
  );
};

SubmitArea.propTypes = {
  onCancel: PropTypes.func.isRequired,
  hasBlankFields: PropTypes.bool,
  send: PropTypes.string.isRequired,
  selection: PropTypes.string,
  loading: PropTypes.bool,
  documentUploaded: PropTypes.bool.isRequired
};

SubmitArea.defaultProps = {
  selection: '',
  loading: false,
  hasBlankFields: false,
};

export default SubmitArea;
