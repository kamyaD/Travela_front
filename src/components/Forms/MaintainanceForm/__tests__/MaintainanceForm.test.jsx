import React from 'react';
import sinon from 'sinon';
import MaintainceForm from '../index';
import SubmitArea from '../FormFieldsets/SubmitArea';

describe('<MaintainanceForm />' , () =>{
  let wrapper, onSubmit;
  onSubmit = jest.fn();

  const props = {
    isLoading: false,
    id: '1',
    status: false,
    history: {
      push: jest.fn()
    },
    errors: [],
    shouldOpen: false,
    onNotificationToggle: jest.fn(),
    addmaintenanceRecord: jest.fn(() => {}),
    updateMaintenanceRecord: jest.fn(() => {}),
    closeModal: jest.fn(() => {}),
    showId: jest.fn(() => {}),
    onChange: jest.fn(() => {}),
    size:10,
    modalType : '',
    managers: [
      {
        fullName: 'Test User',
        email: 'test.user@andela.com'
      }
    ],
    editMaintenance : { maintenance: {} },
    maintenance: {}
  };


  beforeEach(() => {
    wrapper = shallow(<MaintainceForm {...props} />);
  });

  it('renders correctly', () => {
    const mountWrapper = mount(<MaintainceForm {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('calls addmaintenanceRecord when form is submitted', () =>{
    const shallowRender = shallow(<MaintainceForm {...props} />);
    shallowRender.setState({
      reason: 'Broken windows',
      maintainceStart: '10/04/2018',
      maintainceEnd: '11/04/2018'
    });
    const event = {
      preventDefault: jest.fn(),
    };
    sinon.spy(shallowRender.instance(), 'submitMaintainanceData');
    shallowRender.instance().submitMaintainanceData(event);
    expect(shallowRender.instance().submitMaintainanceData.calledOnce).toEqual(true);
  });

  it('calls submitEditedMaintenanceData when edit form is submitted', () =>{
    const currentProps = { ...props, modalType: 'edit-maintenance'};
    const shallowRender = shallow(<MaintainceForm {...currentProps} />);
    shallowRender.setState({
      reason: 'Broken windows',
      maintainceStart: '10/04/2018',
      maintainceEnd: '11/04/2018'
    });
    const event = {
      preventDefault: jest.fn(),
    };
    sinon.spy(shallowRender.instance(), 'submitEditedMaintenanceData');
    shallowRender.instance().submitEditedMaintenanceData(event);
    expect(shallowRender.instance().submitEditedMaintenanceData.calledOnce).toEqual(true);
  });

  it('clears the form when cancel button is clicked', () => {
    const shallowRender = shallow(<MaintainceForm {...props} />);
    shallowRender.setState({
      reason: 'Broken windows',
      maintainceStart: '10/04/2018',
      maintainceEnd: '11/04/2018'
    });
    sinon.spy(shallowRender.instance(), 'handleClearForm');
    shallowRender.instance().handleClearForm();
    expect(shallowRender.instance().handleClearForm.calledOnce).toEqual(true);
  });

  it('Renders error message if field is empty', ()=>{
    const Mountwrapper = mount(<MaintainceForm {...props} />);
    Mountwrapper.find('input[name="reason"]').simulate('blur');
    Mountwrapper.update();
    expect(Mountwrapper.state().errors.reason).toBe('This field is required');
  });
  it('Renders when maintenance props is supplied', ()=>{
    props.maintenance= { departureDate: '2018/04/10', 
      returnDate: '2018/04/11', 
      reason: 'Broken windows' };
    const Mountwrapper = mount(<MaintainceForm {...props} />);
    
   
    expect(Mountwrapper.state().values.reason).toBe('Broken windows');
    expect(Mountwrapper.state().values.maintainanceStart).toBe('04/10/2018');
    expect(Mountwrapper.state().values.maintainanceEnd).toBe('04/11/2018');
  });

  it('Renders when editting', ()=>{
    props.editMaintenance= {maintenance: {
      start: '04/10/2018',
      end: '04/11/2018',
      reason: 'Broken windows'
    }};
    const Mountwrapper = mount(<MaintainceForm {...props} />);
    
    expect(Mountwrapper.state().values.reason).toBe('Broken windows');
    expect(Mountwrapper.state().values.maintainanceStart).toBe('04/10/2018');
    expect(Mountwrapper.state().values.maintainanceEnd).toBe('04/11/2018');
  });
});

describe('<SubmitArea />', () => {
  let wrapper;
  const props = {
    onCancel: jest.fn(),
    handleDelete: jest.fn()
  };

  beforeEach(() => {
    wrapper = mount(<SubmitArea {...props} />);
  });

  it('should call the onCancel prop when click is simulated', () => {
    const onCancel = jest.fn;
    wrapper.find('#cancel').simulate('click');
    expect(onCancel).toBeCalled;
  });

  it('should call the handleDelete prop when click is simulated', () => {
    const handleDelete = jest.fn;
    wrapper.find('#submit').simulate('click');
    expect(handleDelete).toBeCalled;
  });
  
});

