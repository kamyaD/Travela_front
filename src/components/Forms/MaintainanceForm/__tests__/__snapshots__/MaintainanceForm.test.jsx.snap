// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaintainanceForm /> renders correctly 1`] = `
<FormContext
  errors={Object {}}
  targetForm={
    MaintainceForm {
      "context": Object {},
      "defaultState": Object {
        "errors": Object {},
        "hasBlankFields": true,
        "hideNotificationPane": true,
        "hideOverlay": false,
        "hideSideBar": false,
        "openSearch": false,
        "selectedLink": "settings page",
        "values": Object {
          "maintainanceEnd": "",
          "maintainanceStart": "",
          "reason": "",
        },
      },
      "handleClearForm": [Function],
      "handleMaintenanceValues": [Function],
      "handleSetState": [Function],
      "props": Object {
        "addmaintenanceRecord": [Function],
        "closeModal": [Function],
        "editMaintenance": Object {
          "maintenance": Object {},
        },
        "errors": Array [],
        "guestHouseId": "",
        "history": Object {
          "push": [Function],
        },
        "id": "1",
        "isLoading": false,
        "maintenance": Object {},
        "managers": Array [
          Object {
            "email": "test.user@andela.com",
            "fullName": "Test User",
          },
        ],
        "modalType": "",
        "onChange": [Function],
        "onNotificationToggle": [Function],
        "shouldOpen": false,
        "showId": [Function],
        "size": 10,
        "status": false,
        "timelineDateRange": Array [],
        "updateMaintenanceRecord": [Function],
      },
      "refs": Object {},
      "setState": [Function],
      "state": Object {
        "errors": Object {},
        "hasBlankFields": true,
        "hideNotificationPane": true,
        "hideOverlay": false,
        "hideSideBar": false,
        "openSearch": false,
        "selectedLink": "settings page",
        "values": Object {
          "maintainanceEnd": "",
          "maintainanceStart": "",
          "reason": "",
        },
      },
      "submitEditedMaintenanceData": [Function],
      "submitMaintainanceData": [Function],
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": ReactShallowRenderer {
          "_context": Object {},
          "_element": <MaintainceForm
            addmaintenanceRecord={[Function]}
            closeModal={[Function]}
            editMaintenance={
                    Object {
                            "maintenance": Object {},
                          }
            }
            errors={Array []}
            guestHouseId=""
            history={
                    Object {
                            "push": [Function],
                          }
            }
            id="1"
            isLoading={false}
            maintenance={Object {}}
            managers={
                    Array [
                            Object {
                              "email": "test.user@andela.com",
                              "fullName": "Test User",
                            },
                          ]
            }
            modalType=""
            onChange={[Function]}
            onNotificationToggle={[Function]}
            shouldOpen={false}
            showId={[Function]}
            size={10}
            status={false}
            timelineDateRange={Array []}
            updateMaintenanceRecord={[Function]}
    />,
          "_forcedUpdate": false,
          "_instance": [Circular],
          "_newState": Object {
            "errors": Object {},
            "hasBlankFields": true,
            "hideNotificationPane": true,
            "hideOverlay": false,
            "hideSideBar": false,
            "openSearch": false,
            "selectedLink": "settings page",
            "values": Object {
              "maintainanceEnd": "",
              "maintainanceStart": "",
              "reason": "",
            },
          },
          "_rendered": <FormContext
            errors={Object {}}
            targetForm={[Circular]}
            validatorName="validate"
            values={
                    Object {
                            "maintainanceEnd": "",
                            "maintainanceStart": "",
                            "reason": "",
                          }
            }
    >
            <form
                    className="maintainance-form"
                    onSubmit={[Function]}
            >
                    <MaintainanceFieldSets
                            editMaintenance={
                                    Object {
                                            "maintenance": Object {},
                                          }
                            }
                            hasBlankFields={true}
                            values={
                                    Object {
                                            "maintainanceEnd": "",
                                            "maintainanceStart": "",
                                            "reason": "",
                                          }
                            }
                    />
                    <div
                            className="maintainence-line"
                    />
                    <div
                            className="maintainence-submit-area"
                    >
                            <button
                                    className="bg-btn bg-btn--inactive btn-cancel"
                                    onClick={[Function]}
                                    type="button"
                            >
                                    Cancel
                            </button>
                            <button
                                    className="bg-btn bg-btn--active"
                                    disabled={true}
                                    id="submit"
                                    type="submit"
                            >
                                    Save Changes
                            </button>
                    </div>
            </form>
    </FormContext>,
          "_rendering": false,
          "_updater": [Circular],
        },
      },
      "validate": [Function],
      Symbol(enzyme.__setState__): [Function],
    }
  }
  validatorName="validate"
  values={
    Object {
      "maintainanceEnd": "",
      "maintainanceStart": "",
      "reason": "",
    }
  }
>
  <form
    className="maintainance-form"
    onSubmit={[Function]}
  >
    <MaintainanceFieldSets
      editMaintenance={
        Object {
          "maintenance": Object {},
        }
      }
      hasBlankFields={true}
      values={
        Object {
          "maintainanceEnd": "",
          "maintainanceStart": "",
          "reason": "",
        }
      }
    />
    <div
      className="maintainence-line"
    />
    <div
      className="maintainence-submit-area"
    >
      <button
        className="bg-btn bg-btn--inactive btn-cancel"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
      <button
        className="bg-btn bg-btn--active"
        disabled={true}
        id="submit"
        type="submit"
      >
        Save Changes
      </button>
    </div>
  </form>
</FormContext>
`;
